### ISSUE TITLE
For MySQL/Maria, it should use prepare/execute instead of query by default

### ISSUE THREAD

**Issue by testn** [sequelize/sequelize#13543]
## Issue Creation Checklist

[X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

For query without bind/replacement, it should use prepare/execute rather than query for faster performance, especially for mysql/maria

### Additional context

Add any other context or screenshots about the issue here.

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [X] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [X] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
I disagree. Creating one-off prepared statements will result in worse performance.

We need to provide a proper API for prepared statements that devs can use to build the query only once, then execute as many times as they wish, but creating them and throwing them away ourselves [would be wasteful](https://github.com/sequelize/sequelize/issues/13972)

**Comment by ephys** [comment]
Closing in favor of #3495
